### PR TITLE
fix: docker images, pylint issues, exceptions logging

### PR DESCRIPTION
**Pull request by i-savynikh**

### CODE REVIEW

File: _universum/modules/artifact_collector.py [hunk 1]
```diff
@@ -99,28 +99,27 @@ def __init__(self, *args, **kwargs):
...
         try:
             file_name = self.make_file_name(name)
 
-            # File system interaction modules only work with encoded unicode strings
             if isinstance(file_name, six.text_type):
-                encoded_name = file_name.encode("utf-8")
-            else:
-                encoded_name = str(file_name)
+                file_name = str(file_name)
+            assert str == type(file_name)
```
Comment (miltolstoy) [inline _universum/modules/artifact_collector.py]: isinstance?

Comment (i-savynikh) [reply _universum/modules/artifact_collector.py]: no, I want exact checking here

File: tests/test_api.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ def test_git_file_diff(universum_runner):
     with open(os.path.join(universum_runner.artifact_dir, "output.json")) as f:
         result = json.load(f)
 
-    assert result[0]["action"] == "delete"
+    assert result[0]["action"] == "delete", result
```
Comment (i-keliukh) [inline tests/test_api.py]: Please add a comment in the code explaining why this change is needed

File: _universum/modules/artifact_collector.py (hunk 1 above)
Comment (i-keliukh) [reply _universum/modules/artifact_collector.py]: Please discuss using asserts and don't resolve the conversation

File: _universum/lib/module_arguments.py [hunk 3]
```diff
@@ -18,16 +18,17 @@ def __setattr__(self, name, value):
...
     def __getattr__(self, name):
-        if '.' in name:
-            group, name = name.split('.', 1)
-            try:
-                ns = self.__dict__[group]
-            except KeyError:
-                raise AttributeError("No attribute '" + name + "' in module arguments")
-            return getattr(ns, name)
-        else:
+        if not '.' in name:
```
[the remark this answers is not in the document]
Comment (i-savynikh) [reply _universum/lib/module_arguments.py]: I've fixed pylint issue (*unnecessary else after return*)

File: _universum/lib/utils.py [hunk 4]
```diff
@@ -78,15 +74,17 @@ def detect_environment():
 
 
 def create_driver(local_factory, teamcity_factory, jenkins_factory, default=None):
-    if default:
-        env_type = default
-    else:
+    env_type = default
```
[the remark this answers is not in the document]
Comment (i-savynikh) [reply _universum/lib/utils.py]: fix pylint issue and simplify logic

File: analyzers/pylint.py [hunk 5]
```diff
@@ -50,20 +50,16 @@ def execute(self):
...
+            cmd = sh.Command(f"python{self.settings.version}")
             issues = cmd("-m", "pylint", "-f", "json", "--rcfile=" + self.settings.rcfile, *files).stdout
         except sh.CommandNotFound as e:
             sys.stderr.write("No such file or command as '" + str(e) + "'. "
                              "Make sure, that required code report tool is installed.\n")
-        except Exception as e:
+        except Exception as e: #FIXME: handle by this way is possible with 'subprocess.run' usage instead of 'sh'
             if e.stderr and not e.stdout:
-                sys.stderr.write(e.stderr)
+                sys.stderr.write(str(e))
```
[the remark this answers is not in the document]
Comment (i-savynikh) [reply analyzers/pylint.py]: I don't know. Python 3 *Exception* class have no *stderr* and *stdout* attributes. That's why I leave *FIXME* label. 
This chunk should be rewritten using subprocess instead sh. Subprocess result provide possibility to handle cases in required manner

File: analyzers/pylint.py [hunk 6]
```diff
@@ -79,7 +75,7 @@ def execute(self):
             if issues_loads:
                 return 1
         except ValueError as e:
-            sys.stderr.write(e.message)
+            sys.stderr.write(str(e))
```
[the remark this answers is not in the document]
Comment (i-savynikh) [reply analyzers/pylint.py]: I don't know

File: tests/test_argument_check.py [hunk 7]
```diff
@@ -31,7 +31,7 @@ def create_settings(test_type, vcs_type):
         settings.TeamcityServer.user_id = "TeamCityUser"
         settings.TeamcityServer.passwd = "TeamCityPassword"
 
-    if vcs_type in ("git", "gerrit", "github",):
+    if vcs_type in ["git", "gerrit", "github"]:
```
Comment (miltolstoy) [inline tests/test_argument_check.py]: Why you changed tuple to list?

**Comment by i-keliukh** [comment]
One common request to do after porting to python 3: check all issues and update them accordingly, if possible. For example, this one: #280.

File: tests/test_argument_check.py (hunk 7 above)
Comment (i-keliukh) [reply tests/test_argument_check.py]: Because I think it is a little bit cleaner

File: _universum/modules/vcs/base_vcs.py [hunk 8]
```diff
@@ -37,7 +36,8 @@ def clean_sources(self):
         try:
             shutil.rmtree(self.settings.project_root)
         except OSError as e:
-            text = six.text_type(e) + "\nPossible reasons of this error:" + \
+            text = "{}\n".format(e)
+            text += "\nPossible reasons of this error:" + \
```
Comment (k-dovgan) [inline _universum/modules/vcs/base_vcs.py]: Maybe also use fstrings?

File: _universum/modules/vcs/gerrit_vcs.py [hunk 9]
```diff
@@ -48,8 +48,7 @@ def run_ssh_command(self, line, stdin=None, stdout=None):
         try:
             return self.ssh(line, _in=stdin, _out=stdout)
         except sh.ErrorReturnCode as e:
-            text = "Got exit code " + six.text_type(e.exit_code) + \
-                   " while executing the following command:\n" + six.text_type(e.full_cmd)
+            text = "Got exit code {} while executing the following command:\n{}".format(e.exit_code, e.full_cmd)
```
Comment (k-dovgan) [inline _universum/modules/vcs/gerrit_vcs.py]: Maybe also use fstrings?

File: _universum/modules/vcs/perforce_vcs.py [hunk 10]
```diff
@@ -107,7 +107,7 @@ def disconnect(self):
             self.p4.disconnect()
             if not w:
                 return
-            if "Not connected" in w[0].message.message:
+            if "Not connected" in str(w[0].message):
```
Comment (k-dovgan) [inline _universum/modules/vcs/perforce_vcs.py]: Please check: this is a Perforce-specific warning, I'm not sure str() result is the same as .message

File: tests/docker/universum_test_env/Dockerfile [hunk 11]
```diff
@@ -1,20 +1,20 @@
-FROM ubuntu:xenial
+FROM ubuntu:bionic
```
Comment (k-dovgan) [inline tests/docker/universum_test_env/Dockerfile]: Why?

Comment (i-savynikh) [reply tests/docker/universum_test_env/Dockerfile]: because we need python3.7 which is missed in *xenial* apt repository by default

File: _universum/modules/vcs/perforce_vcs.py (hunk 10 above)
Comment (i-savynikh) [reply _universum/modules/vcs/perforce_vcs.py]: *w[0].message* is exception object. This object have no *message* property anymore
